web_infrastructure/jboss_cli.py

```diff
@@ -0,0 +1,133 @@
+            - The host running JBoss.
+        required: false
+        default: '127.0.0.1'
+    port:
+        description:
+            - The port of the associated host running JBoss.
+        required: false
+        default: '9999'
+    jbosshome:
+        description:
+            - The JBoss path environment variable.
+        required: false
+        default: '/usr/share/jbossas/'
+    command:
+        description:
+            - The command to manage the JBoss server.
+        required: true
+'''
+EXAMPLES = '''
+Example playbook entries using the jboss_cli module to manage JBoss.
+
+    tasks:
+
+    - name: set max-threads to a given value
+      jboss_cli: command="/subsystem=ejb3/thread-pool=default:write-attribute(name=max-threads,value=3000)"
+
+    - name: get all configuration and runtime details
+      jboss_cli:
+        command: ":read-resource(include-runtime=true, recursive=true, recursive-depth=10)"
+
+    - name: deploy a given war
+      jboss_cli:
+        command: "deploy /path-up-to/your-app.war"
+'''
+RETURN = '''
+msg:
+    description: execution results from the jboss-cli
+    returned: success, failure
+    type: string
+    sample: outcome => success
+'''
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            command   = dict(required=True, type='str'),
+            user      = dict(required=False, type='str', default=None),
+            password  = dict(required=False, type='str', default=None, no_log=True),
+            host      = dict(required=False, type='str', default='127.0.0.1'),
+            port      = dict(required=False, type='str', default='9999'),
+            jbosshome = dict(required=False, type='str', default='/usr/share/jbossas/'),
+        ),
+    )
+
+    command   = module.params['command']
+    user      = module.params.get('user')
+    password  = module.params.get('password')
+    host      = module.params.get('host')
+    port      = module.params.get('port')
+    jbosshome = module.params.get('jbosshome')
+    # TODO: Add support for --force
+    # TODO: Add support for --file
+    # TODO: Add support for --properties
+    # TODO: Add support for --server-groups
+
+    if os.path.isdir(jbosshome):
+        jbosscli_bin = module.get_bin_path('jboss-cli.sh', True, [jbosshome + '/bin/'])
+    else:
+        module.fail_json(msg='The path ' + jbosshome + ' is not a valid system dir.')
+
+    cmd = [jbosscli_bin]
+
+    cmd.append('--connect')
+
+    if user and password:
+        cmd.append('--user=' + user)
+        cmd.append('--password=' + password)
+
+    cmd.append('--controller=' + host + ':' + port)
+    cmd.append('command=' + command)
+
+    try:
+        rc, stdout, stderr = module.run_command(cmd)
+        # TODO: Prettify output; currently JSON similar, but not JSON compatible
+        module.exit_json(changed=True, stdout=stdout)
+    except subprocess.CalledProcessError as cpe:
+        module.fail_json(msg=str(dir(cpe)))
```

Comment:
This was copied from the existing `system/alternatives.py` which I used as a template. Not sure why it was done exactly that way, but I figured it would be better to copy what is already there than introducing new ways.